pkg/controller/compliancescan/scan.go

```diff
@@ -355,37 +355,131 @@ func newScanPodForNode(scanInstance *compv1alpha1.ComplianceScan, node *corev1.N
 	}
 }
 
+// Add a scanner pod to the scan instance
+// based on scannerType
+func addScannerContainer(scanInstance *compv1alpha1.ComplianceScan, pod *corev1.Pod) *corev1.Pod {
+	falseP := false
+	trueP := true
+
+	switch scanInstance.Spec.ScannerType {
+	case compv1alpha1.ScannerTypeCEL:
+		pod.Spec.Containers = append(pod.Spec.Containers, corev1.Container{
+			Name:  CELScannerContainerName,
+			Image: utils.GetComponentImage(utils.OPERATOR),
+			Command: []string{
+				"compliance-operator", "cel-scanner",
+				// "--api-resource-dir=" + PlatformScanDataRoot,
+				// This is not used for now, we are fetching the data from the API server directly
+				"--profile=" + scanInstance.Spec.Profile,
+				"--check-resultdir=" + "/reports",
+				"--enable-ccr-generation=" + "true", // TODO(@Vincent056): make this configurable
+				"--scan-type=" + "Platform",
+				"--platform=" + os.Getenv("PLATFORM"),
+				"--debug=" + fmt.Sprintf("%t", scanInstance.Spec.Debug),
+				// TODO(@Vincent056): make this configurable, we need to update this to
+				// fetch this value dynamically when we start supporting CEL in Rule CRs
+				"--tailoring=" + "true",
+				"--scan-name=" + scanInstance.Name,
+				"--namespace=" + scanInstance.Namespace,
+			},
+			SecurityContext: &corev1.SecurityContext{
+				AllowPrivilegeEscalation: &falseP,
+				ReadOnlyRootFilesystem:   &trueP,
+				Capabilities: &corev1.Capabilities{
+					Drop: []corev1.Capability{"ALL"},
+				},
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceMemory: resource.MustParse("50Mi"),
+					corev1.ResourceCPU:    resource.MustParse("10m"),
+				},
+				// NOTE: when changing the default limits, remember to also change the
+				// doc text in the CRD.
+				Limits: *scanLimits(scanInstance, "500Mi", "100m"),
+			},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      "tmp-dir",
+					MountPath: "/tmp",
+				},
+				{
+					Name:      "fetch-results",
+					MountPath: PlatformScanDataRoot,
+				},
+				{
+					Name:      "report-dir",
+					MountPath: "/reports",
+				},
+			},
+		})
+	default:
```

Comment:
Another good example of something we can pull out into a separate PR since it helps us organize our existing code in a backwards compatible way and provides a nice seam for us to add the CEL case above.